fondant/component.py

```diff
@@ -38,33 +49,71 @@ def from_file(
             path: Path to the component spec file
         """
         component_spec = ComponentSpec.from_file(path)
-        return cls(component_spec)
+        args_dict = vars(cls._add_and_parse_args(component_spec))
+        user_arguments = cls.get_user_arguments(args_dict, component_spec)
```

Comment:
I think we can drop this function if we just pop all the Fondant arguments before passing the args_dict along.

Reply:
yeah that makes sense